tools/swift-inspect/Sources/SwiftInspectLinux/ElfFile.swift

```diff
@@ -0,0 +1,184 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import LinuxSystemHeaders
+
+class ElfFile {
+  public enum Error: Swift.Error {
+    case FileOpenFailure(_ filePath: String)
+    case FileReadFailure(_ filePath: String, offset: UInt64, size: UInt64)
+    case FileNotElfFormat(_ filePath: String)
+    case MalformedElfFile(_ filePath: String, description: String = "")
+  }
```

Comment:
Please use `ELF` instead of `Elf`. However, some of these feel like they can be replaced with Foundation error values?

Reply:
I will change `Elf` -> `ELF` in the errors. I plan to keep the `Elf` naming convention for data structures to match definitions in `elf.h`. Let me know if you disagree with that.

More generally, I made my own error types because I didn't see general purpose ones. Digging a bit further now, I still only see `NSError`, `CFError`, and `CocoaError`, none of which seems fit for purpose here. I could probably use `POSIXError` in a few places. Am I overlooking something obvious?